### PR TITLE
Added support for ECIES with curve ed25519

### PR DESCRIPTION
**Pull request by timtutt**
Adding support for ECIES. Making Use of CryptoJS AES library for random
IV, Salt generation and AES 256. Updated Readme with example. Added
ecies-test.js

### CODE REVIEW

File: lib/cryptojs/aes.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
```
Comment (indutny) [inline lib/cryptojs/aes.js]: I wonder if it could be installed using npm dependencies?

**Comment by indutny** [comment]
Oh, actually! I just realized that ECIES support does not require any changes to the library's core, am I right?

If so - I think it will be much better to publish it as a separate module! What are your thoughts on this?

**Comment by Kagami** [comment]
Btw, example of ECIES implementation using `elliptic` and `WebCryptoAPI`: https://github.com/bitchan/eccrypto/blob/531148466283acfa1a464832149d833381246c2c/browser.js#L113-L141

**Comment by indutny** [comment]
Ok, @timtutt thank you very much for contribution, but I think it would be much more useful for everyone to publish it as a separate module with dependency to `elliptic`. Please do not hesitate to post a link here or open a PR for README.md with a link to your project. (@Kagami I'd be pleased to accept same PR from you ;) )

**Comment by timtutt** [comment]
@indutny - Sure makes sense to me. Will shoot a PR with the changes pulled
out into a separate module later today.

**Pull request closed without merging**